docs/file-reference/index.md

```diff
@@ -0,0 +1,647 @@
+---
+title: Odo Devfile Reference
+
+language_tabs:
+  - yaml
+
+toc_footers:
+  - <a href='https://odo.dev'>Odo Homepage</a>
+  - <a href='https://devfile.github.io/devfile/index.html'>Devfile GitHub Page</a>
+
+search: true
+---
+
+# Introduction
+
+> A full odo Devfile example using Spring Boot
+
+```yaml
+ 
+schemaVersion: 2.0.0
+metadata:
+  name: springBoot
+  version: 1.0.0
+projects:
+  - name: springbootproject
+    git:
+      location: "https://github.com/odo-devfiles/springboot-ex.git"
+components:
+  - container:
+      name: tools
+      image: maysunfaisal/springbootbuild
+      memoryLimit: 768Mi
+      command: ['tail']
+      args: [ '-f', '/dev/null']
+      mountSources: true
+      volumeMounts:
+        - name: springbootpvc
+          path: /data
+  - container:
+      name: runtime
+      image: maysunfaisal/springbootruntime
+      memoryLimit: 768Mi
+      command: ['tail']
+      args: [ '-f', '/dev/null']
+      endpoints:
+        - name: '8080/tcp'
+          configuration:
+            public: true
+          targetPort: 8080
+      mountSources: false
+      volumeMounts:
+        - name: springbootpvc
+          path: /data
+commands:
+  - exec:
+      id: devBuild
+      commandLine: "/artifacts/bin/build-container-full.sh"
+      component: tools
+      workingDir: /projects/springbootproject
+      group:
+        kind: build
+        isDefault: true
+  - exec:
+      id: devRun
+      commandLine: "/artifacts/bin/start-server.sh"
+      component: runtime
+      workingDir: /
+      group:
+        kind: run
+        isDefault: true
+```
+
+> Download the application
+
+```sh
+$ odo create nodejs --downloadSource
```

Comment:
Should be `--starter` now with https://github.com/openshift/odo/pull/3425 merged